src/decorator.rb

```diff
@@ -0,0 +1,10 @@
+require './nameable'
+
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+end
```

Comment:
> - [ ] Good job so far @fatima-najafi, However, in accordance with the project requirements, you should implement the correct_name method, which returns the result of the correct_name method of @nameable. This part is pending, so kindly proceed with its implementation. It will contribute to your progress in the project.👍
> 
> ![st](https://github.com/fatima-najafi/oop-school-library/assets/40009354/6571e296-aad4-490c-9fdf-9ea438e19735)
> 
> You can use this below as a reference:
>```ruby
> def correct_name
 >     @nameable.correct_name
 >   end
>```